src/workers/runGraph.ts

```diff
@@ -1,20 +1,30 @@
+import { GraphPostLayout, GraphPreLayout } from '@/models/layout'
+
 // eslint-disable-next-line import/default
 import RunGraphWorker from '@/workers/runGraph.worker?worker'
 
-export type ClientMessage = ClientApplicationInitializationMessage
-export type WorkerMessage = WorkerHelloWorldMessage
+export type ClientMessage = ClientLayoutMessage
+export type WorkerMessage = WorkerLayoutMessage
 
-export type ClientApplicationInitializationMessage = {
-  type: 'ping',
+export type ClientLayoutMessage = {
+  type: 'layout',
+  layout: GraphPreLayout,
 }
 
-export type WorkerHelloWorldMessage = {
-  type: 'pong',
+export type WorkerLayoutMessage = {
+  type: 'layout',
+  layout: GraphPostLayout,
 }
 
-export interface IRunGraphWorker extends Omit<Worker, 'postMessage'> {
-  postMessage: (command: ClientMessage, transfer: Transferable[]) => void,
+export interface IRunGraphWorker extends Omit<Worker, 'postMessage' | 'onmessage'> {
+  postMessage: (command: ClientMessage, transfer?: Transferable[]) => void,
   onmessage: ((this: Worker, ev: MessageEvent<WorkerMessage>) => void) | null,
 }
 
-export const worker: IRunGraphWorker = new RunGraphWorker()
+export function getLayoutWorker(onmessage: IRunGraphWorker['onmessage']): IRunGraphWorker {
```

Comment:
Little abstraction so we can have multiple workers. 1:1 for each run. So when we get to sub flows we don't have to worry about overwhelming a single worker.

Reply:
I like this abstraction a lot, I might steal this for use elsewhere